### ISSUE TITLE
kubemark-5: build timed out

### ISSUE THREAD

**Issue by lavalamp** [kubernetes/kubernetes#25849]
https://console.cloud.google.com/storage/kubernetes-jenkins/logs/kubernetes-kubemark-5-gce/4524/

Big gap in timestamps:

```
16:44:08 Generating certs for alternate-names: IP:104.155.134.108,IP:10.0.0.1,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.cluster.local,DNS:kubemark-5-kubemark-master
16:44:11 Warning: Permanently added '104.155.134.108' (ECDSA) to the list of known hosts.
16:44:11 Warning: Permanently added '104.155.134.108' (ECDSA) to the list of known hosts.
16:44:15 Warning: Permanently added '104.155.134.108' (ECDSA) to the list of known hosts.
16:44:15 Unable to find image 'gcr.io/google_containers/etcd:2.2.1' locally
16:44:15 Pulling repository gcr.io/google_containers/etcd
16:44:15 fbea2d67e633: Pulling image (2.2.1) from gcr.io/google_containers/etcd
16:44:15 fbea2d67e633: Pulling image (2.2.1) from gcr.io/google_containers/etcd, endpoint: https://gcr.io/v1/
16:44:15 fbea2d67e633: Pulling dependent layers
16:44:15 bf0f46991aed: Pulling metadata
16:44:16 bf0f46991aed: Pulling fs layer
16:44:16 bf0f46991aed: Download complete
16:44:16 3d5bcd78e074: Pulling metadata
16:44:16 3d5bcd78e074: Pulling fs layer
17:35:32 + rc=124
17:35:32 + [[ 124 -ne 0 ]]
17:35:32 + [[ -x kubernetes/cluster/log-dump.sh ]]
17:35:32 + [[ 124 -eq 124 ]]
17:35:32 
+ echo 'Build timed out'
17:35:32 Build timed out
17:35:32 + echo 'Exiting with code: 124'
17:35:32 Exiting with code: 124
17:35:32 + exit 124
17:35:42 Process leaked file descriptors. See http://wiki.jenkins-ci.org/display/JENKINS/Spawning+processes+from+build for more information
17:35:42 Build step 'Execute shell' marked build as failure
17:35:42 Recording test results
```

**Comment by fejta** [comment]
Assigning to @gmarek per kubemark

**Comment by wojtek-t** [comment]
This is not related to kubemark. It seems like infrastructure problem.

**Comment by gmarek** [comment]
It seems that the test hang on starting etcd on master machine (docker run ...).

**Comment by gmarek** [comment]
We don't have logs from master machine, so it's impossible to debug.

**Comment by fejta** [comment]
@spxtr any idea why/whether `[[ -x kubernetes/cluster/log-dump.sh ]]` failed?

**Comment by spxtr** [comment]
Yes, I just realized that the dockerized e2e don't mount the Jenkins workspace and so that file doesn't exist. They only mount `_artifacts`. @ixdy, should we mount the whole workspace?

**Comment by ixdy** [comment]
I have a slight preference to mounting only the files/directories we explicitly need, since that helps eliminate any implicit dependencies on Jenkins.

**Comment by ixdy** [comment]
Though shouldn't `kubernetes/cluster/log-dump.sh` be in the kubernetes tarball?

**Comment by ixdy** [comment]
Oh, I think I see the issue.

The kubernetes tarball is extracted inside the docker container, which gives us kubernetes/cluster/log-dump.sh. This call is happening on timeout (so outside docker), so the file is inaccessible.

We have a similar issue with all dockerized e2e - no cluster logs on timeout.

We can try moving the timeout handling inside dockerized e2e; this is probably the direction we want to go in eventually anyway.

**Comment by fejta** [comment]
Agree with putting as much logic as possible into the container instead of jenkins/yaml/etc

**Comment by ixdy** [comment]
So, there are two things being discussed here:
a) the kubemark build timed out, seemingly because docker checked out
b) cluster logs aren't collected on time outs

We should probably move b) into a separate issue in kubernetes/test-infra. Have we seen kubemark time out more than once? I'm inclined to believe it's docker being flaky.

**Comment by ixdy** [comment]
kubernetes/test-infra/issues/79 for log dump issue.

**Comment by gmarek** [comment]
I agree - is that the lack of logs makes the debugging impossible, so this is blocked on kubernetes/test-infra#79

**Comment by fejta** [comment]
@ixdy @spxtr any progress around moving timeout logic inside the container?

**Comment by spxtr** [comment]
No progress, haven't been prioritizing it.

**Comment by ixdy** [comment]
The timeout logic is covered by kubernetes/test-infra#79.

I'm inclined to close this issue, because it seemingly hasn't recurred, and there isn't enough data available to figure out what caused docker to get stuck in the first place.

**Comment by spxtr** [comment]
Agreed.
